### PR TITLE
feat: animation update prop hook

### PR DESCRIPTION
**Pull request by MykhailoButkevych-MR**

### CODE REVIEW

File: packages/react-native-select-pro/src/components/arrow/index.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+import type { OptionalToRequired } from '../../helpers/types/optional-to-required';
+import { useAnimation } from '../../hooks/use-animation';
+import type { State } from '../../state/types';
+import type { OnPressSelectControlType } from '../../types';
+import type { Select } from '../select';
+
+const arrowImage = require('./../../assets/icons/chevron-down.png');
+
+type ArrowProps = OptionalToRequired<
```
Comment (irekrog) [inline packages/react-native-select-pro/src/components/arrow/index.tsx:14]: export types to `arrow.types.ts`

File: packages/react-native-select-pro/src/components/arrow/index.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import type { ComponentProps } from 'react';
```
Comment (irekrog) [inline packages/react-native-select-pro/src/components/arrow/index.tsx:1]: this component should be name `arrow.tsx`

Comment (irekrog) [inline packages/react-native-select-pro/src/components/arrow/index.tsx:1]: and in the `index.ts` export this component

**Review by kkaplinski — APPROVED** [review]

File: packages/react-native-select-pro/src/__tests__/react-native-select-pro.test.tsx [hunk 3]
```diff
@@ -596,3 +599,20 @@ describe('Select with multi selection and searchable', () => {
         expect(currentPlaceholderText).toBeTruthy();
     });
 });
+
+describe('Arrow', () => {
```
Comment (kkaplinski) [inline packages/react-native-select-pro/src/__tests__/react-native-select-pro.test.tsx:603]: add to arrow.test.tsx in arrow folder

**Pull request merged**